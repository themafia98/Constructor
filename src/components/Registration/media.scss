@media screen and (max-width: 1000px){

    div.RegistrationBox {
        width: 40%;
    }
}

@media screen and(max-width: 760px){

    div.RegistrationBox {
        width: 85%;
    }
}