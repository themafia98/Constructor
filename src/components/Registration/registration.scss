.RegistrationBox {

    margin-top: 20px;
    box-shadow: 0 0 3px;
    padding: 10px;
    p {
        text-align: center;
        font-weight: bold;
        margin: 5px;
    }

    .loginButton {
        cursor: pointer;
        width: 100%;
        border: 1px solid black;
        border-radius: 20px;
        height: 30px;
        font-weight: bold;
        background: rgb(5, 122, 5);
        margin-top: 10px;
        margin-bottom: 10px;
        display: block;
        color: white;
        transition: .3s all linear;
    }
}