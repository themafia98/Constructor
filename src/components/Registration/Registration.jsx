import React from 'react';
import PropTypes from 'prop-types';

import './registration.scss';

import withFirebase from '../../components/firebaseHOC';


class Registration extends React.PureComponent {

    static propTypes = {
        firebase: PropTypes.object.isRequired, /** @firebase class for use firebase functions */
        indexStream: PropTypes.object.isRequired /** @Events stream object */
    }

    state = {
        firebase: this.props.firebase,
        error: null,
    }

    emailRef = null;
    passwordRef = null;

    setRefEmail = (node) => this.emailRef = node;
    setRefPassword = (node) => this.passwordRef = node;

    createUser = event => {

        const email = this.emailRef ? this.emailRef.value : null;
        const password = this.passwordRef ? this.passwordRef.value : null;
        const { firebase } = this.props;

        if (password && email)
        firebase.registration(email, password)
        .then(response => {
            if (!response) throw new Error('Ivaid registration')
            firebase.db.collection("users").doc(response.user.uid).set({
                'projects': [],
                'email': email,
            }).then(() => this.props.indexStream.emit('EventRegistrationCorrect', response));
        })
        .catch((error) => {
        console.error(error.message);
        this.setState({error: error.message});
        });
        else  return this.setState({error: 'invalid user data'});
    };

    render(){
        let  { error } = this.state;
        return (
            <div className = 'RegistrationBox'>
                <div  className = 'RegistrationForm'>
                    <h3>Registration</h3>
                    <form autoComplete = 'off' className = 'RegForm'>
                        { error ?
                            <p className = 'error'>{error}</p>
                            : null
                        }
                        <span>E-mail</span>
                        <input ref = {this.setRefEmail} type = 'text' />
                        <span>Password</span>
                        <input className = 'password' autoComplete = "off" ref = {this.setRefPassword} type = 'password' />
                        <input onClick = {this.createUser}
                                className = 'loginButton'
                                type = 'button'
                                value = 'registration'
                        />
                    </form>
                </div>
            </div>
        )
    }

}
export default withFirebase(Registration);
