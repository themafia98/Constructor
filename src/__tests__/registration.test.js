"use strict";

import React from 'react';
import firebase from '../Firebase/Firebase';
import renderer from 'react-test-renderer';

import Registration from '../components/Registration/Registration';

firebase.saveSession("NONE");

const RegistrationComponent = renderer.create(
        <Registration firebase = {firebase} />
)
test ('registration.js test', function(){

    let snapshotReg = RegistrationComponent.toJSON();
    expect(snapshotReg).toMatchSnapshot();

    const button = RegistrationComponent.root.find(button => button.props.className === 'loginButton');

    button.props.onClick({});

    snapshotReg = RegistrationComponent.toJSON();
    expect(snapshotReg).toMatchSnapshot();
});