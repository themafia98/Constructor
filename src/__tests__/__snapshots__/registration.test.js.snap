// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`registration.js test 1`] = `
<div
  className="RegistrationBox"
>
  <div
    className="RegistrationBox__RegistrationForm"
  >
    <h3>
      Registration
    </h3>
    <form
      autoComplete="off"
      className="RegistrationForm__RegForm"
    >
      <span>
        E-mail
      </span>
      <input
        placeholder="email"
        type="text"
      />
      <span>
        Password
      </span>
      <input
        autoComplete="off"
        className="password"
        placeholder="password"
        type="password"
      />
      <input
        className="RegForm__loginButton"
        onClick={[Function]}
        type="button"
        value="registration"
      />
    </form>
  </div>
</div>
`;

exports[`registration.js test 2`] = `
<div
  className="RegistrationBox"
>
  <div
    className="RegistrationBox__RegistrationForm"
  >
    <h3>
      Registration
    </h3>
    <form
      autoComplete="off"
      className="RegistrationForm__RegForm"
    >
      <p
        className="error"
      >
        invalid user data
      </p>
      <span>
        E-mail
      </span>
      <input
        placeholder="email"
        type="text"
      />
      <span>
        Password
      </span>
      <input
        autoComplete="off"
        className="password"
        placeholder="password"
        type="password"
      />
      <input
        className="RegForm__loginButton"
        onClick={[Function]}
        type="button"
        value="registration"
      />
    </form>
  </div>
</div>
`;
